lib/externalUpdate.js

```diff
@@ -0,0 +1,51 @@
+import personApi from "./personApi";
+import updateOrgchart from "./updateOrgchart";
+import updateSearch from "./updateSearch";
+
+import { logger } from "./config";
+
+class ExternalUpdate {
+  constructor(cfg) {
+    this.personApi = personApi(cfg);
+    this.orgchartService = updateOrgchart(cfg);
+    this.searchService = updateSearch(cfg);
+  }
+
+  handler() {
+    return (req, res) => {
+      ExternalUpdate.getUserId(req.body)
+        .then(userId => this.getAndDistributeUpdate(userId))
+        .then(updatedProfile => {
+          logger.info(
+            `successfully distributed update for ${updatedProfile.userId}`
+          );
+          res.end();
+        })
+        .catch(e => {
+          logger.error(e);
+          res.status(503);
+          res.end();
+        });
+    };
+  }
+
+  async getAndDistributeUpdate(userId) {
+    return this.personApi(userId)
+      .then(updatedProfile => this.searchService(updatedProfile))
+      .then(updatedProfile => this.orgchartService(updatedProfile));
```

Comment:
Are those promises run sequentially because the orgchart should only update if the search update succeeded? Or could they also run in parallel?